### PR TITLE
wip tipset vectors.

### PR DESCRIPTION
**Pull request by raulk**

### CODE REVIEW

File: gen/builders/tipsets.go [hunk 1]
```diff
@@ -0,0 +1,79 @@
...
+// TipsetSeq provides methods to build a sequence of tipsets, such as
+// adding new tipsets and null rounds.
+type TipsetSeq struct {
+	tipsets  []*Tipset
+	epoch    abi.ChainEpoch
+	messages []*ApplicableMessage
+}
+
+type Tipset struct {
+	tss *TipsetSeq
```
Comment (willscott) [inline gen/builders/tipsets.go:21]: why do you need the back reference?

File: gen/builders/tipsets.go [hunk 2]
```diff
@@ -0,0 +1,79 @@
...
+}
+
+// NullRounds enrols as many null rounds as indicated, advancing the epoch by
+// the count.
+func (tss *TipsetSeq) NullRounds(count uint64) {
+	tss.epoch += abi.ChainEpoch(count)
+}
+
+// Block adds a new block to this tipset, produced by the indicated miner, with
+// the supplied wincount, and containing the listed (and previously staged)
```
Comment (willscott) [inline gen/builders/tipsets.go:66]: this comment claims messages are already staged in the TipsetSeq before Block is called, but the method then appends them.
Something is smelly here.
Maybe `TipsetSeq` should have a method that should both add a block to the last tipset, and learn about the messages in that block?

File: gen/builders/tipsets.go (hunk 1 above)
Comment (raulk) [reply gen/builders/tipsets.go:24]: To add the block messages to the messages index, which is later used by the asserter.

File: gen/builders/tipsets.go (hunk 2 above)
Comment (raulk) [reply gen/builders/tipsets.go:82]: By staged I mean that the message is an `ApplicableMessage`, which can only be generated through a `Messages` object. The Tipset do their own management of messages. For example, you can stage 100 messages, but only actually incorporate 10 to blocks, that's totally fine.

**Comment by raulk** [comment]
This is ready. Closing to open a new PR.

**Pull request closed without merging**